rfcs/geometric-primitives.md

```diff
@@ -0,0 +1,349 @@
+
+```rust
+let unit_sphere = Sphere::UNIT;
+// Some ways we could build meshes from primitives
+let sphere_mesh = Mesh::from(unit_sphere);
+let sphere_mesh = unit_sphere.mesh();
+let sphere_mesh: Mesh = unit_sphere.into();
+```
+
+### Bounding Boxes/Volumes
+
+This section is based off of prototyping work done in [bevy_mod_bounding](https://github.com/aevyrie/bevy_mod_bounding).
+
+A number of bounding box types are provided for 2d and 3d use. Some representations of a bounding box are more efficient depending on the use case. Instead of storing all possible values in a component (wasted space) or computing them with a helper function every time (wasted cpu), each representation is provided as a distinct component that can be updated independently. This gives users of Bevy the flexibility to optimize for their use case without needing to write their own incompatible types from scratch. Consider the functionality built on top of bounding such as physics or collisions - because they are all built on the same fundamental types, they can interoperate.
+
+Note that the bounding types presented in this RFC implement their respective `Obb` or `Aabb` trait. Bounding boxes can be represented by a number of different underlying types; by making these traits instead of structs, systems can easily be made generic over these types depending on the situation.
+
+Because bounding boxes are fully defined in world space, this leads to the natural question of how they are kept in sync with their parent. The plan would be to provide a system similar to transform propagation, that would apply an OBB's precomputed `Transform` to its parent's `GlobalTransform`. Further details are more appropriate for a subsequent bounding RFC/PR. The important point to consider is how this proposal provides common types that can be used for this purpose in the future.
+
+### Frustum Culling
+
+This section is based off of prototyping work done in [bevy_frustum_culling](https://github.com/aevyrie/bevy_frustum_culling).
+
+The provided `Frustum` type is useful for frustum culling, which is generally done on the CPU by comparing each frustum plane with each entity's bounding volume.
+
+```rust
+// What frustum culling might look like:
+for bounding_volume in bound_vol_query.iter() {
+    for plane in camera_frustum.planes().iter() {
+        if bounding_volume.outside(plane) {
+            // Cull me!
+            return;
+        }
+    }
+}
+```
+
+This data structure alone does not ensure the representation is valid; planes could be placed in nonsensical positions. To prevent this, the struct's fields should be made private, and constructors and setters should be provided to ensure `Frustum`s can only be initialized or mutated into valid arrangements.
+
+In addition, by defining the frustum as a set of planes, it is also trivial to support oblique frustums.
```

Comment:
What are oblique frustums useful for?

Reply:
Applications: architectural/drafting projections
Games: rendering portals or to improves sense of speed in driving games

See: https://docs.unity3d.com/Manual/ObliqueFrustum.html

Reply:
Cool! Link that in the RFC?